src/minimax_server/router.clj

```diff
@@ -0,0 +1,18 @@
+(ns minimax-server.router
+  (:import [java.util.function Function]
+           [scarvill.httpserver.response ResponseBuilder])
+  (:require [clojure.string :as str]))
+
+(defn response-with-body [body]
+  (.build (.setBody (new ResponseBuilder) (byte-array (map byte (str body))))))
+
+(defn get-current-player [request]
+  (.getParameterValue request "current_player"))
+
+(defn get-board [request]
+  (str/split (.getParameterValue request "board") #","))
+
+(defn minimax-router [get-move]
+  (proxy [Function] []
+    (apply [request]
+           (response-with-body (get-move (get-current-player request) (get-board request))))))
```

Comment:
There isn't much error handling going on here. Seems like malformed params aren't handled at all. I wonder if you need another layer here. Server Jar -> Clojure adapter -> Controller/Handler layer -> Business Logic (minimax). The adapter layer deals with all the nasty mutating of builders and java interop messiness, the controller is just a function from a request to a response, and the business logic would be your actual minimax algorithm.

I guess what I'm saying is it would be nice if you could just write a Ring-style application with handlers and middleware but use your server jar as a backend instead of Jetty.